### PR TITLE
Additional crossvalidation stats

### PR DESCRIPTION
**Pull request by phyden**
The "CV score" output from the crossvalidation function is replaced by additional stats: precision, recall (=sensitivity), specificity and f1 in addition to the default reported mean balanced accuracy.
This also removes the no more required option for the user to specify a metric (not entirely removed to maintain backwards compatibility). 
Not related to this, pandas was added to the requirements (currently used in SHAP)

### CODE REVIEW

File: phenotrex/cli/generic_func.py [hunk 1]
```diff
@@ -83,13 +83,17 @@ def generic_cv(type, genotype, phenotype, folds, replicates, threads, verb, opti
...
     use_groups = groups is not None
     logger.info(f'Running CV...')
-    score_mean, score_sd, misclass = clf.crossvalidate(records=training_records, cv=folds,
+    #TODO change return fields
+    score_mean_sd, misclass = clf.crossvalidate(records=training_records, cv=folds,
                                                        n_replicates=replicates, groups=use_groups,
                                                        n_jobs=threads,
                                                        reduce_features=reduce_features,
                                                        n_features=n_features,
                                                        demote=not verb)
```
Comment (VarIr) [inline phenotrex/cli/generic_func.py:92]: Please check indentation here. Perhaps we can also have one argument per line:
```suggestion
    score_mean_sd, misclass = clf.crossvalidate(
        records=training_records,
        cv=folds,
        n_replicates=replicates,
        groups=use_groups,
        n_jobs=threads,
        reduce_features=reduce_features,
        n_features=n_features,
        demote=not verb,
    )
```

Comment (phyden) [reply phenotrex/cli/generic_func.py:92]: Right, thanks

File: phenotrex/ml/trex_classifier.py [hunk 2]
```diff
@@ -9,26 +9,35 @@
...
 from phenotrex.structure.records import TrainingRecord, GenotypeRecord
 from phenotrex.ml.cccv import CompleContaCV
 from phenotrex.util.helpers import get_x_y_tn_ft, get_groups
 from phenotrex.ml.feature_select import recursive_feature_elimination, DEFAULT_STEP_SIZE, \
     DEFAULT_SCORING_FUNCTION
 
 
+def specificity_score(y, y_pred, **kwargs) -> float:
+    """
+    Method for binary classification to get specificity: recall of the negative class
```
Comment (VarIr) [inline phenotrex/ml/trex_classifier.py:26]: Some nit picking
```suggestion
    Compute specificity. In binary classification this is equivalent to the recall of the negative class.
```

File: phenotrex/ml/trex_classifier.py [hunk 3]
```diff
@@ -9,26 +9,35 @@
...
     Abstract base class of Trex classifier.
     """
     scoring_function_mapping = {
-        'accuracy': accuracy_score,
         'balanced_accuracy': balanced_accuracy_score,
-        'f1': f1_score
+        'f1': f1_score,
+        'precision': precision_score,
+        'recall': recall_score,
+        'specificity': specificity_score
```
Comment (VarIr) [inline phenotrex/ml/trex_classifier.py:40]: It's always nice to have a comma also in the last line. Good for future diffs.
```suggestion
        'specificity': specificity_score,
```

File: phenotrex/ml/trex_classifier.py [hunk 4]
```diff
@@ -45,7 +54,7 @@ def __init__(self, random_state: int = None, verb: bool = False):
         self.random_state = np.random.RandomState(random_state)
         self.random_state_init = random_state
         self.verb = verb
-        self.vectorizer = CustomVectorizer(binary=True, dtype=np.bool, lowercase=False)
```
Comment (VarIr) [inline phenotrex/ml/trex_classifier.py:48]: Does that mean, `CustomVectorizer` is not used anymore at all? In this case, let's remove the class altogether.

**Review by LokiLuciferase — APPROVED** [review]
Thanks, really great that you took the time to do this! LGTM!

**Pull request merged**